### DISCUSSION TITLE
Need a paragraph in List to style

### DISCUSSION THREAD

**Discussion by s-schmitz** [remarkjs/remark#640] [Ideas] [answered]
Hi folks,

I need to style the text of a list with css:
![image](https://user-images.githubusercontent.com/49489072/109133741-9cb58780-7755-11eb-908d-77eec2991a22.png)

But unfortunately there is no [selector](https://stackoverflow.com/questions/5688712/is-there-a-css-selector-for-text-nodes)  to select only the text and not the entire sub-tag.

Is it possible to wrap the text of a list with a paragraph like that:
![image](https://user-images.githubusercontent.com/49489072/109134054-eef6a880-7755-11eb-84d3-4c4dbd0d929d.png)

Thanks in advance

**Comment by wooorm** [comment, marked as answer]
Approach the problem from the other side: style the li's instead

**Reply by indyteo** [reply]
Such an approach would lead him to having the entire sub-lists styled because they're in the outer `<li>`.
I think he rather what to style only the text elements, which seems to be more tricky...
@nonspectare maybe you can try the following:
Apply your style to the `<li>` element, then "reset" it in the `<ul>` child by applying the opposite. Example:
```css
li {
  background-color: red;
}

li ul {
  background-color: white;
}
```
I understand that it isn't perfect, but maybe it can help you!

**Reply by s-schmitz** [reply]
Thanks for your suggestions. I think my description was not sufficient.  I need to style something like this:
![image](https://user-images.githubusercontent.com/49489072/109488941-80338b00-7a86-11eb-9410-94acdab0c2ec.png)

**Reply by ChristianMurphy** [reply]
@nonspectare I believe @wooorm's and @indyteo's suggestions would allow for that, have you given their suggestions a try?

**Reply by s-schmitz** [reply]
@ChristianMurphy Yes of course. Maybe it's a lack of knowledge. But I'm not able to achieve my example above with the `<p>`-tags with the previous suggestions.
Currently, I use the workaround with `##List##` to select the `<strong>` tag of the text.

The intention of the idea was to ask the devs whether it makes scene to add an additional `<p>`-tag, so it's easier to style Markdown lists.

**Reply by ChristianMurphy** [reply]
> But I'm not able to achieve my example above with the `<p>`-tags with the previous suggestions.

:thinking: the suggestions above don't use a `<p>` tag...
They show how to do the same without a `<p>` tag.

> The intention of the idea was to ask the devs whether it makes scene to add an additional <p>-tag, so it's easier to style Markdown lists.

in `remark`? no.
The language follows CommonMark standard https://commonmark.org which specifies that a there is not a paragraph.
Lists don't often contain a full paragraph, so using a `<p>` for each fragment of text would be semantically incorrect, it would be a `<span>` if anything.

However if you want to add a paragraph or span through a plugin, you can. https://unifiedjs.com/learn/ provides context on how abstract syntax trees work and how plugins can be created.

**Reply by s-schmitz** [reply]
@ChristianMurphy Thanks for the detailed explanation. That answered my questions.
